internal/stacks/stackruntime/internal/stackeval/for_each.go

```diff
@@ -69,22 +69,18 @@ func evaluateForEachExpr(ctx context.Context, expr hcl.Expression, phase EvalPha
 	case ty.IsObjectType() || ty.IsMapType():
 		// okay
 
-	case !result.Value.IsKnown():
-		// we can't validate further without knowing the value
-		return result, diags
-
 	case ty.IsSetType():
-		if markSafeLengthInt(result.Value) == 0 {
-			// we are okay with an empty set
-			return result, diags
-		}
-
 		// since we can't use a set values that are unknown, we treat the
 		// entire set as unknown
 		if !result.Value.IsWhollyKnown() {
 			return result, diags
 		}
 
+		if markSafeLengthInt(result.Value) == 0 {
```

Comment:
Question: why are we doing all this validation for Set types, but not for Map/Object types? 🤨 It's not a blocker, since you aren't introducing that split, I'm just confused.

Reply:
Oh wait. I bet it's because we expect `each.key` to be a string, somewhere. Map/obj values always have strings in `each.key` and can have arbitrary crap in `each.value`, but for sets the element type has to be valid in  both slots.

Reply:
Yeah, that's right, all maps / objects are supported, but not all sets